kong/plugins/oauth2/access.lua

```diff
@@ -95,22 +121,21 @@ local function get_redirect_uris(client_id)
                                  load_oauth2_credential_by_client_id_into_memory,
                                  client_id)
     if err then
-      return responses.send_HTTP_INTERNAL_SERVER_ERROR(err)
+      return internal_server_error(err)
     end
   end
   return client and client.redirect_uris or nil, client
 end
 
 local function retrieve_parameters()
   -- OAuth2 parameters could be in both the querystring or body
-  local uri_args = ngx_req_get_uri_args()
-  local method   = ngx_req_get_method()
+  local uri_args = kong.request.get_query()
+  local method   = kong.request.get_method()
 
   if method == "POST" or method == "PUT" or method == "PATCH" then
-    ngx.req.read_body()
-    local body_args = public_utils.get_body_args()
+    local body_args = kong.request.get_body()
```

Comment:
This might give `nil`, making the following statement runtime-crash. See also: https://github.com/Kong/kong/issues/3823

Yes, it was a problem in original too.